libdnf/conf/option_string_list.cpp

```diff
@@ -28,26 +29,34 @@ namespace libdnf {
 OptionStringList::OptionStringList(const ValueType & default_value)
     : Option(Priority::DEFAULT),
       icase(false),
+      delimiters(DEFAULT_DELIMITERS),
       default_value(default_value),
       value(default_value) {}
 
-OptionStringList::OptionStringList(const ValueType & default_value, std::string regex, bool icase)
+OptionStringList::OptionStringList(
+    const ValueType & default_value, std::string regex, bool icase, const char * delimiters)
     : Option(Priority::DEFAULT),
       regex(std::move(regex)),
       icase(icase),
+      delimiters(delimiters),
```

Comment:
Ok I have changed it to use the same approach as `OptionBool` regarding the default delimiters.

In repoquery command I have also used `OptionStringList` directly without `StringListOption` from `session.cpp` as I didn't want to add custom delimiter support there.
In general I am not sure if we need the `StringListOption` wrapper, we use it only sometimes and it could be confusing.

There is still the problem with specifying the options multiple times - it is currently not supported but I think I would prefer to work on that in next PRs.